plugins/optimization-detective/site-health.php

```diff
@@ -0,0 +1,304 @@
+ * This is to avoid checking for REST API availability during a frontend request. Note that when the plugin is first
+ * installed, the 'od_rest_api_unavailable' option will not be in the database, as the check has not been performed
+ * yet. Once Site Health's weekly check happens or when a user accesses the admin so that the admin_init action fires,
+ * then at this point the check will be performed at {@see od_maybe_run_rest_api_health_check()}. In practice, this will
+ * happen immediately after the user activates a plugin since the user is redirected back to the plugin list table in
+ * the admin. The reason for storing the negative unavailable state as opposed to the positive available state is that
+ * when an option does not exist then `get_option()` returns `false` which is the same falsy value as the stored `'0'`.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @return bool Whether unavailable.
+ */
+function od_is_rest_api_unavailable(): bool {
+	return 1 === (int) get_option( 'od_rest_api_unavailable', '0' );
+}
+
+/**
+ * Tests availability of the Optimization Detective REST API endpoint.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @param array<string, mixed>|WP_Error $response REST API response.
+ * @return array{label: string, status: string, badge: array{label: string, color: string}, description: string, actions: string, test: string} Result.
+ */
+function od_compose_site_health_result( $response ): array {
+	$common_description_html = '<p>' . wp_kses(
+		sprintf(
+			/* translators: %s is the REST API endpoint */
+			__( 'To collect URL Metrics from visitors the REST API must be available to unauthenticated users. Specifically, visitors must be able to perform a <code>POST</code> request to the <code>%s</code> endpoint.', 'optimization-detective' ),
+			'/' . OD_REST_API_NAMESPACE . OD_URL_METRICS_ROUTE
+		),
+		array( 'code' => array() )
+	) . '</p>';
+
+	$result = array(
+		'label'       => __( 'Optimization Detective\'s REST API endpoint is available', 'optimization-detective' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => __( 'Optimization Detective', 'optimization-detective' ),
+			'color' => 'blue',
+		),
+		'description' => $common_description_html . '<p><strong>' . esc_html__( 'This appears to be working properly.', 'optimization-detective' ) . '</strong></p>',
+		'actions'     => '',
+		'test'        => 'optimization_detective_rest_api',
+	);
+
+	$error_label            = __( 'Optimization Detective\'s REST API endpoint is unavailable', 'optimization-detective' );
+	$error_description_html = '<p>' . esc_html__( 'You may have a plugin active or server configuration which restricts access to logged-in users. Unauthenticated access must be restored in order for Optimization Detective to work.', 'optimization-detective' ) . '</p>';
+
+	if ( is_wp_error( $response ) ) {
+		$result['status']      = 'critical';
```

Comment:
I think we should leave this as `recommended`. `critical` should be reserved for truly critical issues with the site, like security bugs for instance. If we start "watering down" what `critical` means, it will lose its purpose.

I understand it's critical for Optimization Detective to work, but it's not critical for the overall site.
```suggestion
		$result['status']      = 'recommended';
```

Reply:
OK, fair enough. Reverted in b747fd575d17e5b0138d7d6688a6925739901b6c.